File: Python/Template/Call_Func.py
# pip install azure-cognitiveservices-vision-computervision
from azure.cognitiveservices.vision.computervision import ComputerVisionClient
from azure.cognitiveservices.vision.computervision.models import OperationStatusCodes
from msrest.authentication import CognitiveServicesCredentials
import os
import sys
import time

# Authenticate
subscription_key = "YOUR_SUBSCRIPTION_KEY"
endpoint = "YOUR_ENDPOINT"
computervision_client = ComputerVisionClient(endpoint, CognitiveServicesCredentials(subscription_key))

# Analyze an image
image_url = "https://example.com/sample.jpg"

features = ['categories', 'description', 'color']
details = ['celebrities']
description_details = ['captions']

results = computervision_client.analyze_image(image_url, visual_features=features, details=details, description_details=description_details)

if 'request_id' in results.headers:
    # Check for Text in Image
    if 'description' in results.analyze_result:
        print("Image Description:")
        for caption in results.analyze_result['description']['captions']:
            print(f"{caption['text']} (Confidence: {caption['confidence']})")

    # Display Categories
    if 'categories' in results.analyze_result:
        print("Image Categories:")
        for category in results.analyze_result['categories']:
            print(f"{category['name']} (Confidence: {category['score']})")

    # Display Colors
    if 'color' in results.analyze_result:
        print("Dominant Colors:")
        for color in results.analyze_result['color']['dominant_colors']:
            print(color)

    # Detect Celebrities
    if 'celebrities' in results.analyze_result:
        print("Celebrities:")
        for celeb in results.analyze_result['celebrities']:
            print(celeb['name'])

else:
    print("Image analysis failed. Status Code: ", results.status_code)

File: Go/Template/Call_Func.go
// go get github.com/Azure/azure-sdk-for-go
package main

import (
    "context"
    "fmt"
    "github.com/Azure/azure-sdk-for-go/services/cognitiveservices/v2.1/computervision"
    "github.com/Azure/azure-sdk-for-go/services/cognitiveservices/v2.1/computervision/computervision"
    "github.com/Azure/go-autorest/autorest"
)

func main() {
    // Set up the credentials
    apiKey := "YOUR_SUBSCRIPTION_KEY"
    endpoint := "YOUR_ENDPOINT"
    creds := NewCognitiveServicesCredentials(apiKey)
    authorizer := autorest.NewCognitiveServicesAuthorizer(creds)

    // Create a new client
    client := computervision.New(endpoint)
    client.Authorizer = authorizer

    // Analyze an image
    imageURL := "https://example.com/sample.jpg"
    features := []computervision.VisualFeatureTypes{computervision.Description, computervision.Categories, computervision.Color}
    info, err := client.AnalyzeImage(context.Background(), imageURL, features, []computervision.Details{computervision.Celebrities})

    if err != nil {
        fmt.Println("Image analysis failed.")
        return
    }

    // Display the analysis results
    if info.Description != nil {
        fmt.Println("Image Description:")
        for _, caption := range *info.Description.Captions {
            fmt.Printf("%s (Confidence: %.2f)\n", *caption.Text, *caption.Confidence)
        }
    }

    if info.Categories != nil {
        fmt.Println("Image Categories:")
        for _, category := range *info.Categories {
            fmt.Printf("%s (Confidence: %.2f)\n", *category.Name, *category.Score)
        }
    }

    if info.Color != nil {
        fmt.Println("Dominant Colors:")
        for _, color := range *info.Color.DominantColors {
            fmt.Println(*color)
        }
    }

    if info.Celebrities != nil {
        fmt.Println("Celebrities:")
        for _, celeb := range *info.Celebrities {
            fmt.Println(*celeb.Name)
        }
    }
}